README.md

```diff
@@ -5,7 +5,126 @@
+|Package| Download Link                                                                                        | Description                                                                |
+|----|------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+|Workleap.DomainEventPropagation.Abstractions| ![nuget](https://img.shields.io/nuget/v/Workleap.DomainEventPropagation.Abscratcions.svg?logo=nuget) | Contains abstractions that are used for publishing and recieveing messages |
+|Workleap.DomainEventPropagation.Publishing| ![nuget](https://img.shields.io/nuget/v/Workleap.DomainEventPropagation.Publishing.svg?logo=nuget)   | Contains classes for message publishing                                    |
+|Workleap.DomainEventPropagation.Subscription| ![nuget](https://img.shields.io/nuget/v/Workleap.DomainEventPropagation.Subscription.svg?logo=nuget) |  Contains classes for subscriptions                                        |
+
+### What does the Workleap.DomainEventPropagation package do?
+* Provides list of existing Topics
+* Serializes domain events
+* Manages publication fields
+* Provides webhook endpoint (.net core only)
+* Provides IDomainEvent and IDomainEventHandler<> interfaces
+* Deserializes domain events
+* Mediates call to DomainEventHandlers
+* Validates configurations
+* Handles topic subscriptions
+
+### Using the Workleap.DomainEventPropagation.Publishing package to publish events
+
+When using dotnet core, you can register event propagation at startup in the service collection.
+```
+// Method 1: Lazily bind the options to a configuration section
+services.AddEventPropagationPublisher();
+services.AddOptions<EventPropagationPublisherOptions>().BindConfiguration(EventPropagationPublisherOptions.SectionName);
+
+// Method 2: Set options values directly in C#
+services.AddEventPropagationPublisher(opt =>
+{
+  opt.TopicName = "<topic_name_to _publish_to>",
+  opt.TopicAccessKey = "<provided from keyVault>",
+  opt.TopicEndpoint = "<azure_topic_uri>"
+});
+```
+Configuration is required. Configuration can be loaded from the appsettings file by passing the IConfiguration instance (see above). The topic access key should be stored securely in a key vault.
+```json
+// appsettings.json
+{
+  "EventPropagation": {
+    "TopicName": "<topic_name_to_publish_to>",
+    "TopicAccessKey": "<keyVault_provided_value>",
+    "TopicEndpoint": "<azure_topic_uri>"
+  }
+}
+```
+To publish an event, use the IEventPropagationClient interface (via injection). Use the PublishDomainEventAsync to publish the event. The required Subject field is a string description to provide context for the event.
+
+```
+var domainEvent = new ExampleDomainEvent
+{
+    Id = Guid.NewGuid().ToString(),
+    Date = DateTime.UtcNow
+};
+
+await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", coolCandorEvent);
+```
```

Comment:
```suggestion
```csharp
var domainEvent = new ExampleDomainEvent
{
    Id = Guid.NewGuid().ToString(),
    Date = DateTime.UtcNow
};

await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", domainEvent);
```
```